starknet_devnet/blueprints/feeder_gateway.py

```diff
@@ -68,8 +64,40 @@ async def get_block():
     else:
         block = state.starknet_wrapper.blocks.get_by_number(block_number)
 
+    return block
+
+@feeder_gateway.route("/get_block", methods=["GET"])
+def get_block():
+    """Endpoint for retrieving a block identified by its hash or number."""
+
+    block_hash = request.args.get("blockHash")
+    block_number = request.args.get("blockNumber", type=custom_int)
+
+    block = _get_block_object(block_hash=block_hash, block_number=block_number)
+
     return Response(block.dumps(), status=200, mimetype="application/json")
 
+@feeder_gateway.route("/get_block_traces", methods=["GET"])
+def get_block_traces():
+    """Returns the traces of the transactions in the specified block."""
+
+    block_hash = request.args.get("blockHash")
+    block_number = request.args.get("blockNumber", type=custom_int)
+
+    block = _get_block_object(block_hash=block_hash, block_number=block_number)
+
+    traces = []
+    for transaction in block.transaction_receipts:
+        tx_hash = hex(transaction.transaction_hash)
+        trace = state.starknet_wrapper.transactions.get_transaction_trace(tx_hash)
+
+        # expected trace is equal to response of get_transaction, but with the hash property
+        trace_dict = trace.dump()
+        trace_dict["transaction_hash"] = tx_hash
+        traces.append(trace_dict)
+
+    return jsonify({ "traces": traces })
```

Comment:
> implementation of response_objects on the server side instead of the test side

Do you mean "checking of implementation", not the implementation itself? Of course we don't implement anything on the test side.

We already apply this pattern elsewhere in the code, I can apply it here as well - we would keep it in tests but also add the `BlockTransactionTraces.loads` in the endpoint as well?

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/181/commits/5e2d8fc80460abc9f6ec084df6dd48e51de59e4d